charts/skypilot/templates/oauth2-proxy-deployment.yaml

```diff
@@ -0,0 +1,51 @@
+{{- if and .Values.ingress.enabled (index .Values.ingress "oauth2-proxy" "enabled") }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ .Release.Name }}-oauth2-proxy
+    skypilot.co/component: oauth2-proxy
+  name: {{ .Release.Name }}-oauth2-proxy
+  namespace: {{ .Release.Namespace }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}-oauth2-proxy
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}-oauth2-proxy
+        skypilot.co/component: oauth2-proxy
+    spec:
+      containers:
+      - name: oauth2-proxy
+        image: quay.io/oauth2-proxy/oauth2-proxy:latest
+        imagePullPolicy: Always
+        args:
+        - --provider=oidc
+        - --oidc-issuer-url={{ index .Values.ingress "oauth2-proxy" "oidc-issuer-url" }}
+        - --email-domain={{ index .Values.ingress "oauth2-proxy" "email-domain" | default "*" }}
+        - --upstream=file:///dev/null
+        - --http-address=0.0.0.0:4180
+        - --skip-provider-button=true
+        - --reverse-proxy=true
+        {{- if (index .Values.ingress "oauth2-proxy" "use-http") }}
+        - --cookie-secure=false
+        {{- end }}
+        env:
+        - name: OAUTH2_PROXY_CLIENT_ID
+          value: {{ index .Values.ingress "oauth2-proxy" "client-id" | quote }}
+        - name: OAUTH2_PROXY_CLIENT_SECRET
+          value: {{ index .Values.ingress "oauth2-proxy" "client-secret" | quote }}
+        - name: OAUTH2_PROXY_COOKIE_SECRET
+          {{- if index .Values.ingress "oauth2-proxy" "cookie-secret" }}
+          value: {{ index .Values.ingress "oauth2-proxy" "cookie-secret" | quote }}
+          {{- else }}
+          # Generate a random cookie secret if not provided
+          value: {{ randAlphaNum 16 | b64enc | quote }}
```

Comment:
Looks like the cookie-secret will be updated every time a `hem upgrade` is called if it is not explicitly specified, will this invalidate all the issued cookies?